Recursively Determine Size of coresymbolicationd Cache When Pruning

chgibb-apple opened the pull request:
#### b16a3adebc84c911d70174523b0a238bfd47bd76
<pre>
Recursively Determine Size of coresymbolicationd Cache When Pruning
<a href="https://bugs.webkit.org/show_bug.cgi?id=267053">https://bugs.webkit.org/show_bug.cgi?id=267053</a>
<a href="https://rdar.apple.com/120428325">rdar://120428325</a>

Reviewed by Alexey Proskuryakov.

In the prune-coresymbolicationd-cache-if-too-large buildbot step, we should recursively walk the top-level directory when determining its size. We should also use sudo when attempting to delete the cache.
Failing to recursively determine size results in size always being 0 for /System/Library/Caches/com.apple.coresymbolicationd. Failing to use sudo prevents deletion of the same.

* Tools/CISupport/build-webkit-org/steps.py:
(PruneCoreSymbolicationdCacheIfTooLarge):
* Tools/Scripts/delete-if-too-large:
(main):

Canonical link: <a href="https://commits.webkit.org/272843@main">https://commits.webkit.org/272843@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2e1c894425975621c79413d81ba921072116194

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33202 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11976 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35116 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35839 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30243 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/34173 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14323 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9146 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29377 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33677 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10105 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29663 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8806 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8956 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29636 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37170 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30172 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30017 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35065 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9087 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7024 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32926 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/33085 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10807 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7717 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9671 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9756 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/delete-if-too-large:
```diff
@@ -25,8 +25,18 @@
+import subprocess
 from pathlib import Path
 
+def get_folder_size(folder):
```
Would it be reasonable to shell out to `du` instead? We're already shelling out to `rm` for deleting anyway, not using native Python code.

aproskuryakov commented on Tools/Scripts/delete-if-too-large:
```diff
@@ -41,14 +41,18 @@ def main():
-            os.remove(directory)
+        tenGigabytesInKiloBytes = 1024 * 1024  * 10
+        
+        #directorySize will be in kilobytes
```
Please update comment style - there should be a space after "#", and it should be a full sentence with a period at the end.

aproskuryakov commented on Tools/Scripts/delete-if-too-large:
```diff
@@ -41,14 +41,18 @@ def main():
+        directorySize = int(subprocess.check_output(['sudo', 'du', '-sk', directoryPath]).decode().strip().split()[0])
+
+        if directorySize >= tenGigabytesInKiloBytes:
+            print(f'{directory} is {directorySize}. Deleting') 
```
This log line won't read well when printed, because it lacks units for size (and doesn't even explain that the number is a size), and lacks a period at the end.

aproskuryakov reviewed (approved):
This is fine to land as is (with just the below nitpicks addressed). However, it would probably be better to only delete old cached files, keeping those that have been used recently.

chgibb-apple committed:
Recursively Determine Size of coresymbolicationd Cache When Pruning
https://bugs.webkit.org/show_bug.cgi?id=267053
rdar://120428325

Reviewed by Alexey Proskuryakov.

In the prune-coresymbolicationd-cache-if-too-large buildbot step, we should recursively walk the top-level directory when determining its size. We should also use sudo when attempting to delete the cache.
Failing to recursively determine size results in size always being 0 for /System/Library/Caches/com.apple.coresymbolicationd. Failing to use sudo prevents deletion of the same.

* Tools/CISupport/build-webkit-org/steps.py:
(PruneCoreSymbolicationdCacheIfTooLarge):
* Tools/Scripts/delete-if-too-large:
(main):

Canonical link: https://commits.webkit.org/272843@main
